sql/core/src/main/scala/org/apache/spark/sql/execution/Exchange.scala

```diff
@@ -488,6 +488,12 @@ private[sql] case class EnsureRequirements(sqlContext: SQLContext) extends Rule[
   }
 
   def apply(plan: SparkPlan): SparkPlan = plan.transformUp {
+    case operator @ Exchange(partitioning, child, _) =>
+      child.children match {
+        case Exchange(childPartitioning, baseChild, _)::Nil =>
```

Comment:
Yes, I thought the same, but then it will again be not as generic as this, since SparkStrategies are applied first and till that time we don;t have the exchanges added. So it will be similar to my previous change done in optimizer in that it will check that the child plan is an aggregate or not instead of testing for an Exchange. Will that be acceptable ?